lib/api/workloadendpoint.go

```diff
@@ -29,7 +31,8 @@ type WorkloadEndpoint struct {
 type WorkloadEndpointMetadata struct {
 	unversioned.ObjectMetadata
 
-	// The name of the endpoint.
+	// The name of the endpoint.  This may be omitted on a create, in which case an endpoint
+	// ID will be automatically created, and the endpoint ID will be included in the response.
```

Comment:
Ok, cool.  So long as that's the plan I'm happy for it.  Is there an issue open?